sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/CosmosCatalog.scala

```diff
@@ -65,12 +65,9 @@ class CosmosCatalog
     */
   override def initialize(name: String,
                           options: CaseInsensitiveStringMap): Unit = {
-    val accountConfig = CosmosAccountConfig.parseCosmosAccountConfig(
-      options.asCaseSensitiveMap().asScala.toMap)
-    this.client = new CosmosClientBuilder()
-      .key(accountConfig.key)
-      .endpoint(accountConfig.endpoint)
-      .buildAsyncClient()
+    val config = options.asCaseSensitiveMap().asScala.toMap
+
+    this.client = CosmosClientCache(CosmosClientConfiguration(config, useEventualConsistency = true), None)
```

Comment:
CosmosConfigEntry has infra for supporting default value, could you please rely on that instead of passing the default explicitly here?

see sample please:
https://github.com/moderakh/azure-sdk-for-java/blob/eb13c04777dd8fb183f30e4b217e6c3bba573dcc/sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala#L102

Reply:
Done